logger/logger.go

```diff
@@ -2,56 +2,67 @@ package logger
 
 import (
 	"flag"
+	"os"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	logrus_cloudwatchlogs "github.com/kdar/logrus-cloudwatchlogs"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
 )
 
-// Log is an instance of the global zap.Logger
-var Log *zap.Logger
-var logLevel zapcore.Level
+// Log is an instance of the global logrus.Logger
+var Log *logrus.Logger
+var logLevel logrus.Level
 
 // InitLogger initializes the Entitlements API logger
-func InitLogger() *zap.Logger {
-	if Log == nil {
-		viper.SetDefault("INGRESS_LOG_LEVEL", "INFO")
-		viper.AutomaticEnv()
-		switch viper.GetString("INGRESS_LOG_LEVEL") {
-		case "DEBUG":
-			logLevel = zapcore.DebugLevel
-		case "ERROR":
-			logLevel = zapcore.ErrorLevel
-		default:
-			logLevel = zapcore.InfoLevel
-		}
-		if flag.Lookup("test.v") != nil {
-			logLevel = zapcore.FatalLevel
-		}
+func InitLogger() *logrus.Logger {
 
-		cfg := zapcore.EncoderConfig{
-			TimeKey:        "ts",
-			LevelKey:       "level",
-			NameKey:        "logger",
-			CallerKey:      "caller",
-			MessageKey:     "msg",
-			LineEnding:     zapcore.DefaultLineEnding,
-			EncodeLevel:    zapcore.LowercaseLevelEncoder,
-			EncodeTime:     zapcore.EpochTimeEncoder,
-			EncodeDuration: zapcore.SecondsDurationEncoder,
-			EncodeCaller:   zapcore.ShortCallerEncoder,
-		}
+	viper.SetDefault("INGRESS_LOG_LEVEL", "INFO")
+	viper.SetDefault("INGRESS_LOG_GROUP", "platform-dev")
+	viper.SetDefault("INGRESS_LOG_STREAM", "platform")
+	key := viper.GetString("INGRESS_CW_AWS_ACCESS_KEY_ID")
+	secret := viper.GetString("INGRESS_CW_AWS_SECRET_ACCESS_KEY")
+	group := viper.GetString("INGRESS_LOG_GROUP")
+	stream := viper.GetString("INGRESS_LOG_STREAM")
+	viper.AutomaticEnv()
+	switch viper.GetString("INGRESS_LOG_LEVEL") {
+	case "DEBUG":
+		logLevel = logrus.DebugLevel
+	case "ERROR":
+		logLevel = logrus.ErrorLevel
+	default:
+		logLevel = logrus.InfoLevel
+	}
+	if flag.Lookup("test.v") != nil {
+		logLevel = logrus.FatalLevel
+	}
 
-		logger, _ := zap.Config{
-			Encoding:         "json",
-			Level:            zap.NewAtomicLevelAt(logLevel),
-			OutputPaths:      []string{"stdout"},
-			ErrorOutputPaths: []string{"stderr"},
-			EncoderConfig:    cfg,
-		}.Build()
+	Log = &logrus.Logger{
+		Out:   os.Stdout,
+		Level: logLevel,
+	}
 
-		defer logger.Sync()
-		Log = logger
+	formatter := &logrus.JSONFormatter{
+		FieldMap: logrus.FieldMap{
+			logrus.FieldKeyTime:  "time",
+			logrus.FieldKeyFunc:  "caller",
+			logrus.FieldKeyLevel: "level",
+			logrus.FieldKeyMsg:   "message",
```

Comment:
Should this be `msg`?

Reply:
It wasn't in the old logger, but we can change it.